teachertool/src/components/DebouncedInput.tsx

```diff
@@ -0,0 +1,28 @@
+import { useRef } from "react";
+import { Input, InputProps } from "react-common/components/controls/Input";
+
+export interface DebouncedInputProps extends InputProps {
+    intervalMs?: number | undefined;
+}
+
+// This functions like the React Common Input, but debounces onChange calls
+// so they only fire once every `interval` milliseconds (defined in props)
+export const DebouncedInput: React.FC<DebouncedInputProps> = props => {
+    const timerId = useRef<NodeJS.Timeout | undefined>(undefined);
+
+    const onChangeDebounce = (newValue: string) => {
+        if (timerId.current) {
+            clearTimeout(timerId.current);
+        }
+
+        timerId.current = setTimeout(() => {
```

Comment:
What is the expected behavior if the component unmounts while a timeout is pending?

Reply:
Hmm good question. I'll add some code to fire the onChange immediately if we're umounting and the timeout is pending.